src/uct/base/uct_md.c

```diff
@@ -368,17 +368,94 @@ static ucs_status_t uct_mem_check_flags(unsigned flags)
     return UCS_OK;
 }
 
-ucs_status_t uct_md_mem_alloc(uct_md_h md, size_t *length_p, void **address_p,
-                              unsigned flags, const char *alloc_name, uct_mem_h *memh_p)
+ucs_status_t uct_mem_alloc_check_params(const uct_mem_alloc_params_t *params)
 {
+    const uct_alloc_method_t *method;
     ucs_status_t status;
 
-    status = uct_mem_check_flags(flags);
+    if (!(params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_LENGTH_PTR)) {
+        ucs_debug("The length value for allocating memory isn't set: %s",
+                  ucs_status_string(UCS_ERR_INVALID_PARAM));
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if ((params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_LENGTH_PTR) &&
+        (*params->length_p == 0)) {
+        ucs_debug("The length value for allocating memory is set to zero: %s",
+                  ucs_status_string(UCS_ERR_INVALID_PARAM));
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if ((params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_FLAGS) &&
+        (params->flags & UCT_MD_MEM_FLAG_FIXED)) {
+        if (!(params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_ADDR_PTR)) {
+            ucs_debug("UCT_MD_MEM_FLAG_FIXED requires setting of"
+                      " UCT_MEM_ALLOC_PARAM_FIELD_ADDR_PTR field");
+            return UCS_ERR_INVALID_PARAM;
+        } else {
+            if (!*params->address_p || ((uintptr_t)*params->address_p % ucs_get_page_size())) {
+                ucs_debug("UCT_MD_MEM_FLAG_FIXED requires valid page size aligned address");
+                return UCS_ERR_INVALID_PARAM;
+            }
+        }
+	}
+
+    if (!(params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_METHODS)) {
+        ucs_debug("Allocation methods field not set");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    if ((params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_METHODS) &&
```

Comment:
why need to check params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_METHODS again?

Reply:
https://github.com/openucx/ucx/pull/5562/files/85a64823fb26a88471539edf511dad03b3204c12#diff-6282d9e471573ebace2884f6debada46R403 checks if FIELD_METHODS is set whereas https://github.com/openucx/ucx/pull/5562/files/85a64823fb26a88471539edf511dad03b3204c12#diff-6282d9e471573ebace2884f6debada46R408 checks if methods.count is 0 once it's confirmed that FIELD_METHODS is set. I'll combine these two with else if assuming that is the suggestion.